doc/how_to/migrate/from_streamlit.md

```diff
@@ -0,0 +1,950 @@
+        func = calculation_b
+
+    time_start = time.perf_counter()
+    result = func()
+    time_end = time.perf_counter()
+    calculation_input.disabled=calculation_input.loading=run_input.disabled=run_input.loading=False
+    yield f"""Done!
+
+Result: {result}
+
+The function took {time_end - time_start:1.1f} seconds to complete"""
+
+
+calculation_input = pn.widgets.RadioBoxGroup(name="Calculation", options=["A", "B"])
+run_input = pn.widgets.Button(name="Press to run calculation", icon="caret-right", button_type="primary", sizing_mode="fixed", width=250)
+
+pn.Column(
+    "Which calculation would you like to perform?",
+    calculation_input,
+    pn.bind(calculation_message, calculation_input),
+    run_input,
+    pn.bind(calculation_result, run_input, calculation_input)
+).servable()
+```
+
+The app looks like
+
+![Panel Multiple Updates Alternative Example](https://user-images.githubusercontent.com/42288570/243402005-0e520ed0-a41f-4339-a854-14ff31cb04ce.gif)
+
+### Multiple Results Example
+
+Sometimes you want to output multiple results individually as soon as they are ready.
+
+This is for example the case for Large (AI) Language Models that generates one token after the other.
+
+#### Streamlit Multiple Results Example
+
+```python
+import random
+import time
+
+import streamlit as st
+
+run = st.button("Run model")
+
+def model():
+    time.sleep(1)
+    return random.randint(0, 100)
+
+if not run:
+    st.write("The model has not run yet")
+else:
+    with st.spinner("Running..."):
+        for i in range(0,10):
+            result = model()
+            st.write(f"Result {i}: {result}")
+```
+
+The app looks like
+
+![Streamlit Multiple Results Example](https://user-images.githubusercontent.com/42288570/243421817-bcd0d8d4-89af-4ef4-8924-f20a8d7a8d33.gif)
+
+#### Panel Multiple Results Example
+
+With Panel you will use a *generator function* to display multiple results from code execution as soon as they are ready.
+
+```python
+import random
+import time
+
+import panel as pn
+
+pn.extension(sizing_mode="stretch_width", template="bootstrap")
+
+run_input = pn.widgets.Button(name="Run model")
+
+def model():
+    time.sleep(1)
+    return random.randint(0, 100)
+
+def results(running):
+    if not running:
+        yield "Calculation did not run yet"
+        return # This will break the execution
+
+    run_input.disabled=True
+    loading_indicator = pn.Row(pn.indicators.LoadingSpinner(value=True, width=50, height=50, align="center"), "Running... Please wait!", align="center")
+    layout = pn.Column(loading_indicator)
+    yield layout # This will display the layout
+
+    for i in range(0,10):
+        result = model()
+        message = f"Result {i}: {result}"
+        layout.append(message)
+        yield layout # This will force the UI to update
+
+    layout.pop(0)
+    run_input.disabled=False
+    yield layout # This will force the UI to update
+
+
+pn.Column(run_input, pn.bind(results, run_input)).servable()
```

Comment:
Wrap try/finally
```suggestion
import random
import time
import panel as pn

pn.extension(sizing_mode="stretch_width")
run_input = pn.widgets.Button(name="Run model")


def model():
    time.sleep(1)
    return random.randint(0, 100)


def results(running):
    if not running:
        yield "Calculation did not run yet"
        return  # This will break the execution
    try:
        run_input.disabled = True
        loading_indicator = pn.Row(
            pn.indicators.LoadingSpinner(
                value=True, width=50, height=50, align="center"
            ),
            "Running... Please wait!",
            align="center",
        )
        layout = pn.Column(loading_indicator)
        yield layout  # This will display the layout
        for i in range(0, 10):
            result = model()
            message = f"Result {i}: {result}"
            layout.append(message)
            yield layout
        layout.pop(0)
    finally:
        run_input.disabled = False
    yield layout  # This will force the UI to update


pn.Column(run_input, pn.bind(results, run_input)).servable()
```